src/threading.c

```diff
@@ -287,8 +287,8 @@ void jl_pgcstack_getkey(jl_get_pgcstack_func **f, jl_pgcstack_key_t *k)
 #endif
 
 jl_ptls_t *jl_all_tls_states JL_GLOBALLY_ROOTED;
-uint8_t *jl_measure_compile_time = NULL;
-uint64_t *jl_cumulative_compile_time = NULL;
+volatile uint8_t jl_measure_compile_time = 0;
+volatile uint64_t jl_cumulative_compile_time = 0;
```

Comment:
volatile makes no useful promise here
```suggestion
uint8_t jl_measure_compile_time = 0;
uint64_t jl_cumulative_compile_time = 0;
```